src/transforms/aws_ec2_metadata.rs

```
@@ -281,66 +294,37 @@ impl MetadataClient {
         Ok(token)
     }
 
-    pub async fn get_document(&mut self) -> Result<Option<IdentityDocument>, crate::Error> {
-        let token = self.get_token().await?;
-
-        let mut parts = self.host.clone().into_parts();
-        parts.path_and_query = Some(DYNAMIC_DOCUMENT.clone());
-        let uri = Uri::from_parts(parts)?;
-
-        let req = Request::get(uri)
-            .header(TOKEN_HEADER.as_ref(), token.as_ref())
-            .body(Body::empty())?;
-
-        let res = self.client.send(req).await?;
-
-        if res.status() != StatusCode::OK {
-            warn!(message="Identity document request failed.", status = %res.status());
-            return Ok(None);
-        }
-
-        let body = body_to_bytes(res.into_body()).await?;
-
-        serde_json::from_slice(&body[..])
-            .map_err(Into::into)
-            .map(Some)
+    pub async fn get_document(&mut self) -> Option<IdentityDocument> {
+        self.get_document_error()
```

You may also want to not swallow the error, so that the caller can act on it.